Formula/jam.rb

```diff
@@ -17,4 +17,23 @@ def install
     system "make", "CC=#{ENV.cc}", "CFLAGS=#{ENV.cflags}", "LOCATE_TARGET=bin"
     bin.install "bin/jam", "bin/mkjambase"
   end
+
+  test do
+    (testpath/"Jamfile").write <<-EOS.undent
+      Main jamtest : jamtest.c ;
+    EOS
+
+    (testpath/"jamtest.c").write <<-EOS.undent
+      #include <stdio.h>
+
+      int main(void)
+      {
+          printf("Jam Test\\n");
+          return 0;
+      }
+    EOS
+
+    system bin/"jam"
+    system "./jamtest"
```

Comment:
Unfortunately, `jam` exits with 0 even for some failed runs, so we can't use the exit code as a test that it did its job correctly.